psi4/src/psi4/lib3index/dfhelper.cc

```diff
@@ -3107,6 +3223,71 @@ void DFHelper::compute_J(std::vector<SharedMatrix> D, std::vector<SharedMatrix>
         }
     }
 }
+void DFHelper::compute_J_combined(std::vector<SharedMatrix> D, std::vector<SharedMatrix> J, double* Mp, double* T1p, double* T2p, std::vector<std::vector<double>>& D_buffers, size_t bcount, size_t block_size) {
+//    double* coul_metp = metric_prep_core(1.0);
+    double ZERO[16] = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
+    double* zerop = &ZERO[0];
+    double* coul_metp;
+    std::unique_ptr<double[]> metric;
+
+
+    if (!hold_met_) {
+        metric = std::unique_ptr<double[]>(new double[naux_ * naux_]);
+        coul_metp = metric.get();
+        std::string filename = return_metfile(1.0);
+        get_tensor_(std::get<0>(files_[filename]), coul_metp, 0, naux_ - 1, 0, naux_ - 1);
+    } else
+        coul_metp = metric_prep_core(1.0);
+
+
+    for (size_t i = 0; i < J.size(); i++) {
+        double* Dp = D[i]->pointer()[0];
+        double* Jp = J[i]->pointer()[0];
+        C_DCOPY(naux_ *  nthreads_, zerop, 0, T1p, 1);
+        C_DCOPY(naux_, zerop, 0, T2p, 1);
```

Comment:
In these two lines you're copying `naux_ * nthreads_` and `naux_` elements from the **16** element ZERO array.  Are you simply trying to zero out `T1p` and `T2p`? If so, use `memset(T1p, 0, sizeof(double) * naux_ * nthreads_` for the first one and something similar for `T2p`.

Reply:
Yes exactly, and ok.